primitives/npos-elections/compact/src/lib.rs

```diff
@@ -194,20 +194,19 @@ fn struct_def(
 				all_edges
 			}
 
-			fn unique_targets(&self) -> Vec<Self::Target> {
+			fn unique_targets(&self) -> _npos::sp_std::prelude::Vec<Self::Target> {
 				// NOTE: this implementation returns the targets sorted, but we don't use it yet per
 				// se, nor is the API enforcing it.
-				let mut all_targets: Vec<Self::Target> = Vec::with_capacity(self.average_edge_count());
+				use _npos::sp_std::collections::btree_set::BTreeSet;
+
+				let mut all_targets: BTreeSet<Self::Target> = BTreeSet::new();
```

Comment:
I don't remember, we do have proper test for this `unique_targets`, right?

Reply:
No, AFAICT this crate's macros don't generate any test functions, and we don't test them with an example case. Would you like me to add that?

Reply:
hmm let me see. There must be some tests in the root level crate, otherwise I've been a very bad coder.

Reply:
https://github.com/paritytech/substrate/blob/master/primitives/npos-elections/src/tests.rs#L1141

Reply:
There are some instances of `uniqueue_targets` there, I presume that back then I found them enough, but would be good if you ensure they are safe and sound again.

Reply:
Oh cool! I'll look over them there.

Reply:
IMO [`unique_targets_len_edge_count_works`](https://github.com/paritytech/substrate/blob/354176941e29f5da533b389879c3b64af5da6b76/primitives/npos-elections/src/tests.rs#L1395-L1459) sufficiently exercises `unique_targets`: in two different instances, it provides non-unique inputs, and validate that `unique_targets` unifies them.